src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -1234,6 +1234,15 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
         tmp = list_api->read_item(plist, pos);
     }
 
+    void visit_ListLen(const ASR::ListLen_t& x) {
+        uint64_t ptr_loads_copy = ptr_loads;
+        ptr_loads = 0;
+        this->visit_expr(*x.m_arg);
+        ptr_loads = ptr_loads_copy;
+        llvm::Value* plist = tmp;
+        tmp = list_api->len(plist);
```

Comment:
`list_len: i32 = len([1.0, 2.0])` for this example, why not use the value from `x.m_value`??
```suggestion
        if (x.m_value) {
            this->visit_expr(*x.m_value);
        } else {
            uint64_t ptr_loads_copy = ptr_loads;
            ptr_loads = 0;
            this->visit_expr(*x.m_arg);
            ptr_loads = ptr_loads_copy;
            llvm::Value* plist = tmp;
            tmp = list_api->len(plist);
        }
```